### PR TITLE
Initial proposal for ICMP forwarding and ECN for flow mode 

### PR DESCRIPTION
**Pull request by gloinul**

### CODE REVIEW

File: draft-kuehlewind-masque-connect-ip.md [hunk 1]
```diff
@@ -681,23 +695,60 @@ in progress, registration and use of Context IDs is left for future work at this
...
-ECN requires coordination with the end-to-end communication points as it should only be
-used if the endpoints are also capable and willing to signal congestion
-notifications to the other end and react accordingly if a congestion
-notification is received. 
+ECN requires coordination with the end-to-end communication points as
+it should only be used if the endpoints are also capable and willing
+to signal congestion notifications to the other end and react
+accordingly if a congestion notification is received.
+
+The probing and verification in the ULP of end-to-end ECN requires per
```
Comment (mirjak) [inline draft-kuehlewind-masque-connect-ip.md:703]: ULP?

**Review by mirjak — APPROVED** [review]
Looks good to me!

File: draft-kuehlewind-masque-connect-ip.md (hunk 1 above)
Comment (zaheduzzaman) [reply draft-kuehlewind-masque-connect-ip.md:703]: Upper Layer Protocol.

File: draft-kuehlewind-masque-connect-ip.md [hunk 2]
```diff
@@ -681,23 +695,60 @@ in progress, registration and use of Context IDs is left for future work at this
...
+there are multiple matching use the IP selector with the most number
+of field that matches fully.
+
+Some messages may be applicable both to the proxy and the client. For
+example an verified ICMPv6 Packet Too Big is applicable both to the
+proxy and the client. Others like ICMPv6 Destiantion Unreachable
+(Type=1), Code=3 (Address unreachable) and Code=4 (Port unreachable)
+is only possible to act on by the client.
+
+QUESTION: Any ICMP messages that should be suppressed by the proxy?
```
Comment (zaheduzzaman) [inline draft-kuehlewind-masque-connect-ip.md:746]: Proxy should not suppress the ICMP messages unless -
- the ICMP messages is meant to be consumed by the proxy
- the ICMP messages is not meant to be forwarded to the particular client.

**Review by zaheduzzaman — COMMENTED** [review]
LGTM, made a PR for some editorial fixes

File: draft-kuehlewind-masque-connect-ip.md (hunk 1 above)
Comment (gloinul) [reply draft-kuehlewind-masque-connect-ip.md:703]:
```suggestion
The probing and verification in the upper layer protocol of end-to-end ECN requires per
```

File: draft-kuehlewind-masque-connect-ip.md (hunk 2 above)
Comment (gloinul) [reply draft-kuehlewind-masque-connect-ip.md:746]:
```suggestion
QUESTION: Which ICMP messages should be suppressed by the proxy?
```

**Pull request merged**